streaming/src/message/message_bundle.h

```diff
@@ -177,5 +183,31 @@ class StreamingMessageBundle : public StreamingMessageBundleMeta {
       const std::list<StreamingMessagePtr> &message_list, uint32_t raw_data_size,
       uint8_t *raw_data);
 };
+
+/// Databundle is super-bundle that contains channel information (upstream
+/// channel id & bundle meta data) and raw buffer pointer.
+struct DataBundle {
+  uint8_t *data = nullptr;
+  uint32_t data_size;
+  ObjectID from;
+  uint32_t last_barrier_id;
+  StreamingMessageBundleMetaPtr meta;
+  bool is_reallocated = false;
```

Comment:
Is `is_reallocated ` variable necessary?

Reply:
yes, only when `is_reallocated=true` will we delete `data`